src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -0,0 +1,897 @@
+
+            if (!node.HasErrors)
+            {
+                var indexer = node.Indexer;
+                CheckInvocationArgMixing(
+                    node.Syntax,
+                    indexer,
+                    node.ReceiverOpt,
+                    indexer.Parameters,
+                    node.Arguments,
+                    node.ArgumentRefKindsOpt,
+                    node.ArgsToParamsOpt,
+                    _localScopeDepth,
+                    _diagnostics);
+            }
+
+            return null;
+        }
+
+        public override BoundNode? VisitFunctionPointerInvocation(BoundFunctionPointerInvocation node)
+        {
+            using var _ = GetArgumentPlaceholders(receiverOpt: null, node.Arguments);
+            base.VisitFunctionPointerInvocation(node);
+
+            if (!node.HasErrors)
+            {
+                var method = node.FunctionPointer.Signature;
+                CheckInvocationArgMixing(
+                    node.Syntax,
+                    method,
+                    receiverOpt: null,
+                    method.Parameters,
+                    node.Arguments,
+                    node.ArgumentRefKindsOpt,
+                    argsToParamsOpt: default,
+                    _localScopeDepth,
+                    _diagnostics);
+            }
+
+            return null;
+        }
+
+        public override BoundNode? VisitAwaitExpression(BoundAwaitExpression node)
+        {
+            var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, uint)>.GetInstance();
+            GetAwaitableInstancePlaceholders(placeholders, node.AwaitableInfo, GetValEscape(node.Expression, _localScopeDepth));
+            using var _ = new PlaceholderRegion(this, placeholders);
+            base.VisitAwaitExpression(node);
+            return null;
+        }
+
+        private void GetAwaitableInstancePlaceholders(ArrayBuilder<(BoundValuePlaceholderBase, uint)> placeholders, BoundAwaitableInfo awaitableInfo, uint valEscapeScope)
+        {
+            if (awaitableInfo.AwaitableInstancePlaceholder is { } placeholder)
+            {
+                placeholders.Add((placeholder, valEscapeScope));
+            }
+        }
+
+        // Based on NullableWalker.VisitDeconstructionAssignmentOperator().
+        public override BoundNode? VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
+        {
+            base.VisitDeconstructionAssignmentOperator(node);
+
+            var left = node.Left;
+            var right = node.Right;
+            var variables = GetDeconstructionAssignmentVariables(left);
+            VisitDeconstructionArguments(variables, right.Syntax, right.Conversion, right.Operand);
+            variables.FreeAll(v => v.NestedVariables);
+            return null;
+        }
+
+        private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> variables, SyntaxNode syntax, Conversion conversion, BoundExpression right)
```

Comment:
Was this implementation moved here from elsewhere?

Reply:
This method and the caller were added in this PR, although based on the corresponding visit methods from `NullableWalker`.